tests/phpunit/tests/test-wpml-config-locations.php

```diff
@@ -0,0 +1,119 @@
+
+				@mkdir( $dest_dir );
+
+				foreach ( glob( $source_dir . '/*.*' ) as $file ) {
+					copy( $file, $dest_dir . '/' . basename( $file ) );
+				}
+			}
+		}
+	}
+
+	public static function wpTearDownAfterClass() {
+		parent::wpTearDownAfterClass();
+
+		// Remove previously copied themes and plugins from wp-content.
+		foreach ( self::$dirs as $type => $subdirs ) {
+			foreach ( $subdirs as $name ) {
+				$dest_dir = WP_CONTENT_DIR . "/{$type}/{$name}";
+
+				foreach ( glob( $dest_dir . '/*.*' ) as $file ) {
+					unlink( $file );
+				}
+
+				rmdir( $dest_dir );
+			}
+		}
+	}
+
+	public function test_in_polylang() {
+		@mkdir( WP_CONTENT_DIR . '/polylang' );
+		copy( dirname( __DIR__ ) . '/data/wpml-config.xml', WP_CONTENT_DIR . '/polylang/wpml-config.xml' );
+
+		$files    = ( new PLL_WPML_Config() )->get_files();
+		$expected = array( 'Polylang' => WP_CONTENT_DIR . '/polylang/wpml-config.xml' );
+
+		unlink( WP_CONTENT_DIR . '/polylang/wpml-config.xml' );
+		rmdir( WP_CONTENT_DIR . '/polylang' );
+
+		$this->assertSameSetsWithIndex( $expected, $files );
+	}
+
+	public function test_in_theme() {
+		switch_theme( 'best-theme' );
+
+		$files    = ( new PLL_WPML_Config() )->get_files();
+		$expected = array( 'themes/best-theme' => get_theme_root() . '/best-theme/wpml-config.xml' );
+
+		$this->assertSameSetsWithIndex( $expected, $files );
+	}
+
+	public function test_in_child_theme() {
+		switch_theme( 'best-child' );
+		$files    = ( new PLL_WPML_Config() )->get_files();
+		$expected = array(
+			'themes/best-theme' => get_theme_root() . '/best-theme/wpml-config.xml',
+			'themes/best-child' => get_theme_root() . '/best-child/wpml-config.xml',
+		);
+
+		$this->assertSameSetsWithIndex( $expected, $files );
+	}
+
+	public function test_in_active_plugin() {
+		activate_plugin( 'best-plugin/best-plugin.php' );
+		$files    = ( new PLL_WPML_Config() )->get_files();
+		$expected = array( 'plugins/best-plugin' => WP_PLUGIN_DIR . '/best-plugin/wpml-config.xml' );
+
+		$this->assertSameSetsWithIndex( $expected, $files );
+	}
+
+	public function test_in_mu_plugin() {
+		@mkdir( WPMU_PLUGIN_DIR );
+		$filename_1 = WPMU_PLUGIN_DIR . '/wpml-config.xml';
+		copy( dirname( __DIR__ ) . '/data/wpml-config.xml', $filename_1 );
+
+		@mkdir( WPMU_PLUGIN_DIR . '/must-use' );
+		$filename_2 = WPMU_PLUGIN_DIR . '/must-use/wpml-config.xml';
+		copy( dirname( __DIR__ ) . '/data/wpml-config.xml', $filename_2 );
+
+		@symlink( dirname( __DIR__ ) . '/data/plugins/best-plugin', WPMU_PLUGIN_DIR . '/best-plugin' );
+
+		$files    = ( new PLL_WPML_Config() )->get_files();
+		$expected = array(
+			'mu-plugins'             => $filename_1,
+			'mu-plugins/must-use'    => $filename_2,
+			'mu-plugins/best-plugin' => WPMU_PLUGIN_DIR . '/best-plugin/wpml-config.xml',
+		);
+
+		unlink( WPMU_PLUGIN_DIR . '/best-plugin' );
+
+		unlink( $filename_2 );
+		rmdir( WPMU_PLUGIN_DIR . '/must-use' );
+
+		unlink( $filename_1 );
+		rmdir( WPMU_PLUGIN_DIR );
```

Comment:
This should be moved to `wpTearDownAfterClass()` with some `file_exist()` IMHO.

Reply:
Same as above. This would be even less readable in this test as we would need to transform file names to properties.